stdlib/builtins.pyi

```diff
@@ -324,12 +329,12 @@ class complex:
     def __add__(self, __x: complex) -> complex: ...
     def __sub__(self, __x: complex) -> complex: ...
     def __mul__(self, __x: complex) -> complex: ...
-    def __pow__(self, __x: complex, mod: None = ...) -> complex: ...
+    def __pow__(self, __x: complex | int | float, mod: None = ...) -> complex: ...
```

Comment:
`complex | float` is enough. Type checkers imagine that `int` is a subclass of `float`, even though it actually isn't.

A protocol involving `SupportsIndex` might be more accurate, if these behave similarly to `math` functions (#6211 and #6216).